h/groups/views.py

```diff
@@ -101,26 +103,21 @@ def _read_group(request, group):
     """
     url = request.route_url('group_read', pubid=group.pubid, slug=group.slug)
 
-    result = search.search(request,
-                           private=False,
-                           params={"group": group.pubid, "limit": 1000})
-    annotations = [presenters.AnnotationHTMLPresenter(
-                       storage.annotation_from_dict(a))
-                   for a in result['rows']]
-
-    # Group the annotations by URI.
-    # Create a dict mapping the (normalized) URIs of the annotated documents
-    # to the most recent annotation of each document.
-    annotations_by_uri = collections.OrderedDict()
-    for annotation in annotations:
-        normalized_uri = uri.normalize(annotation.uri)
-        if normalized_uri not in annotations_by_uri:
-            annotations_by_uri[normalized_uri] = annotation
-            if len(annotations_by_uri) >= 25:
-                break
-
-    document_links = [annotation.document_link
-                      for annotation in annotations_by_uri.values()]
+    documents = (
+        request.db.query(models.Document)
+        .options(joinedload(models.Document.meta))
+        .filter(models.Annotation.shared == expression.true())
```

Comment:
It does seem to work. There's a couple of filters below this line that achieve that mapping:

```python
.filter(models.Annotation.target_uri_normalized == models.DocumentURI.uri_normalized)
.filter(models.DocumentURI.document_id == models.Document.id)
```

It doesn't seem to be necessary for these filters to be stated _before_ the shared annotations filter (and iirc from the days when I used to actually write SQL queries, that rings true to me).